src/devices/src/virtio/block.rs

```diff
@@ -300,27 +294,81 @@ impl Request {
     }
 }
 
-/// Handler that drives the execution of the Block devices
-pub struct BlockEpollHandler {
-    queues: Vec<Queue>,
-    mem: GuestMemory,
+/// Virtio device for exposing block level read/write operations on a host file.
+pub struct Block {
     disk_image: File,
     disk_nsectors: u64,
-    interrupt_status: Arc<AtomicUsize>,
-    interrupt_evt: EventFd,
-    queue_evt: EventFd,
+    avail_features: u64,
+    acked_features: u64,
+    config_space: Vec<u8>,
     rate_limiter: RateLimiter,
+    queues: Option<Vec<Queue>>,
+    mem: Option<GuestMemory>,
+    interrupt_status: Option<Arc<AtomicUsize>>,
+    interrupt_evt: Option<EventFd>,
+    queue_evt: Option<EventFd>,
```

Comment:
I suggest we have the device create and own its `interrupt_evt` and `queue_evt`. On registration the device can provide these events that it want to register instead of having them come from the outside.